app/services/adviser/sign_up.rb

```diff
@@ -0,0 +1,49 @@
+class Adviser::SignUp
+  class AdviserSignUpUnavailableError < RuntimeError; end
+
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attr_reader :application_form
+  attribute :preferred_teaching_subject
+
+  validates :preferred_teaching_subject, inclusion: { in: :teaching_subject_names, allow_blank: false }
+
+  def initialize(application_form, *args)
```

Comment:
In future (maybe current?) versions of Ruby, `*` splats won't capture keyword args, so we need to separately capture those and pass them through to `super`.

`def inititalize(application_form, *args, **kwargs)`